diff --git a/cas-subsystem/src/main/java/org/soulwing/cas/extension/AttributeTransformDefinition.java b/cas-subsystem/src/main/java/org/soulwing/cas/extension/AttributeTransformDefinition.java
index 358eba8..8b86d8d 100644
--- a/cas-subsystem/src/main/java/org/soulwing/cas/extension/AttributeTransformDefinition.java
+++ b/cas-subsystem/src/main/java/org/soulwing/cas/extension/AttributeTransformDefinition.java
@@ -24,7 +24,6 @@ import org.jboss.as.controller.OperationFailedException;
 import org.jboss.as.controller.PathAddress;
 import org.jboss.as.controller.RestartParentResourceAddHandler;
 import org.jboss.as.controller.RestartParentResourceRemoveHandler;
-import org.jboss.as.controller.ServiceVerificationHandler;
 import org.jboss.as.controller.SimpleResourceDefinition;
 import org.jboss.as.controller.registry.ManagementResourceRegistration;
 import org.jboss.dmr.ModelNode;
@@ -102,8 +101,7 @@ public class AttributeTransformDefinition extends SimpleResourceDefinition {
      */
     @Override
     protected void recreateParentService(OperationContext context,
-        PathAddress parentAddress, ModelNode parentModel,
-        ServiceVerificationHandler verificationHandler)
+        PathAddress parentAddress, ModelNode parentModel)
         throws OperationFailedException {
       ProfileService.ServiceUtil.installService(context, parentModel, 
           parentAddress);
@@ -134,8 +132,7 @@ public class AttributeTransformDefinition extends SimpleResourceDefinition {
      */
     @Override
     protected void recreateParentService(OperationContext context,
-        PathAddress parentAddress, ModelNode parentModel,
-        ServiceVerificationHandler verificationHandler)
+        PathAddress parentAddress, ModelNode parentModel)
         throws OperationFailedException {
       ProfileService.ServiceUtil.installService(context, parentModel, 
           parentAddress);
diff --git a/cas-subsystem/src/main/java/org/soulwing/cas/extension/HostnameVerifierDefinition.java b/cas-subsystem/src/main/java/org/soulwing/cas/extension/HostnameVerifierDefinition.java
index 60bbef3..496b7e4 100644
--- a/cas-subsystem/src/main/java/org/soulwing/cas/extension/HostnameVerifierDefinition.java
+++ b/cas-subsystem/src/main/java/org/soulwing/cas/extension/HostnameVerifierDefinition.java
@@ -27,7 +27,6 @@ import org.jboss.as.controller.PathAddress;
 import org.jboss.as.controller.PathElement;
 import org.jboss.as.controller.RestartParentResourceAddHandler;
 import org.jboss.as.controller.RestartParentResourceRemoveHandler;
-import org.jboss.as.controller.ServiceVerificationHandler;
 import org.jboss.as.controller.SimpleAttributeDefinition;
 import org.jboss.as.controller.SimpleAttributeDefinitionBuilder;
 import org.jboss.as.controller.SimpleListAttributeDefinition;
@@ -138,8 +137,7 @@ class HostnameVerifierDefinition extends SimpleResourceDefinition {
      */
     @Override
     protected void recreateParentService(OperationContext context,
-        PathAddress parentAddress, ModelNode parentModel,
-        ServiceVerificationHandler verificationHandler)
+        PathAddress parentAddress, ModelNode parentModel)
         throws OperationFailedException {
       ProfileService.ServiceUtil.installService(context, parentModel, 
           parentAddress);
@@ -187,8 +185,7 @@ class HostnameVerifierDefinition extends SimpleResourceDefinition {
      */
     @Override
     protected void recreateParentService(OperationContext context,
-        PathAddress parentAddress, ModelNode parentModel,
-        ServiceVerificationHandler verificationHandler)
+        PathAddress parentAddress, ModelNode parentModel)
         throws OperationFailedException {
       ProfileService.ServiceUtil.installService(context, parentModel, 
           parentAddress);
diff --git a/cas-subsystem/src/main/java/org/soulwing/cas/extension/ProfileDefinition.java b/cas-subsystem/src/main/java/org/soulwing/cas/extension/ProfileDefinition.java
index 603d132..d0cc9b4 100644
--- a/cas-subsystem/src/main/java/org/soulwing/cas/extension/ProfileDefinition.java
+++ b/cas-subsystem/src/main/java/org/soulwing/cas/extension/ProfileDefinition.java
@@ -27,7 +27,6 @@ import org.jboss.as.controller.OperationContext;
 import org.jboss.as.controller.OperationFailedException;
 import org.jboss.as.controller.PathAddress;
 import org.jboss.as.controller.RestartParentWriteAttributeHandler;
-import org.jboss.as.controller.ServiceVerificationHandler;
 import org.jboss.as.controller.SimpleAttributeDefinition;
 import org.jboss.as.controller.SimpleAttributeDefinitionBuilder;
 import org.jboss.as.controller.SimpleResourceDefinition;
@@ -244,18 +243,19 @@ public class ProfileDefinition extends SimpleResourceDefinition {
      */
     @Override
     protected void performRuntime(OperationContext context,
-        ModelNode operation, ModelNode model,
-        ServiceVerificationHandler verificationHandler,
-        List<ServiceController<?>> newControllers)
+        ModelNode operation, ModelNode model)
         throws OperationFailedException {
 
       ServiceController<?> controller = 
           ProfileService.ServiceUtil.installService(context, model, 
               PathAddress.pathAddress(
                   operation.get(ModelDescriptionConstants.OP_ADDR)));
-      
-      newControllers.add(
-          (ServiceController<?>) controller);
+
+      //,
+      //        ServiceVerificationHandler verificationHandler,
+      //        List<ServiceController<?>> newControllers
+      //newControllers.add(
+      //    (ServiceController<?>) controller);
     }
 
   }
@@ -316,8 +316,7 @@ public class ProfileDefinition extends SimpleResourceDefinition {
      */
     @Override
     protected void recreateParentService(OperationContext context,
-        PathAddress parentAddress, ModelNode parentModel,
-        ServiceVerificationHandler verificationHandler)
+        PathAddress parentAddress, ModelNode parentModel)
         throws OperationFailedException {
       ProfileService.ServiceUtil.installService(context, parentModel, 
           parentAddress);
diff --git a/cas-subsystem/src/main/java/org/soulwing/cas/extension/ProxyChainDefinition.java b/cas-subsystem/src/main/java/org/soulwing/cas/extension/ProxyChainDefinition.java
index c27d316..2b9a18c 100644
--- a/cas-subsystem/src/main/java/org/soulwing/cas/extension/ProxyChainDefinition.java
+++ b/cas-subsystem/src/main/java/org/soulwing/cas/extension/ProxyChainDefinition.java
@@ -24,7 +24,6 @@ import org.jboss.as.controller.OperationFailedException;
 import org.jboss.as.controller.PathAddress;
 import org.jboss.as.controller.RestartParentResourceAddHandler;
 import org.jboss.as.controller.RestartParentResourceRemoveHandler;
-import org.jboss.as.controller.ServiceVerificationHandler;
 import org.jboss.as.controller.SimpleAttributeDefinition;
 import org.jboss.as.controller.SimpleAttributeDefinitionBuilder;
 import org.jboss.as.controller.SimpleListAttributeDefinition;
@@ -121,8 +120,7 @@ class ProxyChainDefinition extends SimpleResourceDefinition {
      */
     @Override
     protected void recreateParentService(OperationContext context,
-        PathAddress parentAddress, ModelNode parentModel,
-        ServiceVerificationHandler verificationHandler)
+        PathAddress parentAddress, ModelNode parentModel)
         throws OperationFailedException {
       ProfileService.ServiceUtil.installService(context, parentModel, 
           parentAddress);
@@ -152,8 +150,7 @@ class ProxyChainDefinition extends SimpleResourceDefinition {
      */
     @Override
     protected void recreateParentService(OperationContext context,
-        PathAddress parentAddress, ModelNode parentModel,
-        ServiceVerificationHandler verificationHandler)
+        PathAddress parentAddress, ModelNode parentModel)
         throws OperationFailedException {
       ProfileService.ServiceUtil.installService(context, parentModel, 
           parentAddress);
diff --git a/cas-subsystem/src/main/java/org/soulwing/cas/extension/SubsystemDefinition.java b/cas-subsystem/src/main/java/org/soulwing/cas/extension/SubsystemDefinition.java
index e5449e5..2467463 100644
--- a/cas-subsystem/src/main/java/org/soulwing/cas/extension/SubsystemDefinition.java
+++ b/cas-subsystem/src/main/java/org/soulwing/cas/extension/SubsystemDefinition.java
@@ -24,7 +24,6 @@ import org.jboss.as.controller.AbstractBoottimeAddStepHandler;
 import org.jboss.as.controller.AbstractRemoveStepHandler;
 import org.jboss.as.controller.OperationContext;
 import org.jboss.as.controller.OperationFailedException;
-import org.jboss.as.controller.ServiceVerificationHandler;
 import org.jboss.as.controller.SimpleResourceDefinition;
 import org.jboss.as.controller.operations.common.GenericSubsystemDescribeHandler;
 import org.jboss.as.controller.registry.ManagementResourceRegistration;
@@ -85,8 +84,7 @@ public class SubsystemDefinition extends SimpleResourceDefinition {
      */
     @Override
     public void performBoottime(OperationContext context, ModelNode operation,
-        ModelNode model, ServiceVerificationHandler verificationHandler,
-        List<ServiceController<?>> newControllers)
+        ModelNode model)
         throws OperationFailedException {
 
       DescriptorDeploymentProcessor.addStepHandler(context);
diff --git a/cas-subsystem/src/main/java/org/soulwing/cas/extension/TransformerDefinition.java b/cas-subsystem/src/main/java/org/soulwing/cas/extension/TransformerDefinition.java
index 96e99ab..a27cdcd 100644
--- a/cas-subsystem/src/main/java/org/soulwing/cas/extension/TransformerDefinition.java
+++ b/cas-subsystem/src/main/java/org/soulwing/cas/extension/TransformerDefinition.java
@@ -24,7 +24,6 @@ import org.jboss.as.controller.OperationFailedException;
 import org.jboss.as.controller.PathAddress;
 import org.jboss.as.controller.RestartParentResourceAddHandler;
 import org.jboss.as.controller.RestartParentResourceRemoveHandler;
-import org.jboss.as.controller.ServiceVerificationHandler;
 import org.jboss.as.controller.SimpleAttributeDefinition;
 import org.jboss.as.controller.SimpleAttributeDefinitionBuilder;
 import org.jboss.as.controller.SimpleMapAttributeDefinition;
@@ -130,8 +129,7 @@ public class TransformerDefinition extends SimpleResourceDefinition {
      */
     @Override
     protected void recreateParentService(OperationContext context,
-        PathAddress parentAddress, ModelNode parentModel,
-        ServiceVerificationHandler verificationHandler)
+        PathAddress parentAddress, ModelNode parentModel)
         throws OperationFailedException {
       ProfileService.ServiceUtil.installService(context, parentModel, 
           parentAddress);
@@ -162,8 +160,7 @@ public class TransformerDefinition extends SimpleResourceDefinition {
      */
     @Override
     protected void recreateParentService(OperationContext context,
-        PathAddress parentAddress, ModelNode parentModel,
-        ServiceVerificationHandler verificationHandler)
+        PathAddress parentAddress, ModelNode parentModel)
         throws OperationFailedException {
       ProfileService.ServiceUtil.installService(context, parentModel, 
           parentAddress);
diff --git a/pom.xml b/pom.xml
index 0fe176f..3a7be5e 100644
--- a/pom.xml
+++ b/pom.xml
@@ -15,7 +15,7 @@
     An extension for Wildfly that supports CAS authentication.
   </description>
   <properties>
-    <version.org.jasig.cas.client>3.4.1</version.org.jasig.cas.client>
+    <version.org.jasig.cas.client>3.5.0</version.org.jasig.cas.client>
     <module.name>cas</module.name>
   </properties>
   <scm>
@@ -70,6 +70,15 @@
     </dependencies>
   </dependencyManagement>
   <profiles>
+    <profile>
+      <id>wildfly14</id>
+      <properties>
+        <java.version>1.8</java.version>
+        <version.org.wildfly>14.0.1.Final</version.org.wildfly>
+        <version.org.wildfly.core>6.0.2.Final</version.org.wildfly.core>
+        <version.io.undertow>2.0.13.Final</version.io.undertow>
+      </properties>
+    </profile>
     <profile>
       <id>wildfly10</id>
       <properties>
